src/twisted/web/test/test_http.py

```diff
@@ -3406,16 +3406,75 @@ def cookieValue(key, value):
     def test_addCookieSameSite(self):
         """
         L{http.Request.setCookie} supports a C{sameSite} argument.
+        It will set the cookie with samesite=lax and samesite=strict values.
         """
         self._checkCookie(b"foo=bar; SameSite=lax", b"foo", b"bar", sameSite="lax")
         self._checkCookie(b"foo=bar; SameSite=lax", b"foo", b"bar", sameSite="Lax")
         self._checkCookie(
             b"foo=bar; SameSite=strict", b"foo", b"bar", sameSite="strict"
         )
 
-        self.assertRaises(
+    def test_addCookieSameSiteNoneSecure(self):
+        """
+        L{http.Request.setCookie} supports the C{sameSite} argument together with secure.
+        It will set the cookie with samesite=none and secure attributes to be sure that
+        all values of samesite are supported. secure attribute is a necessary condition
+        if the samesite is none
+        """
+        self._checkCookie(
+            b"foo=bar; Secure; SameSite=none",
+            b"foo",
+            b"bar",
+            sameSite="None",
+            secure=True,
+        )
+        self._checkCookie(
+            b"foo=bar; Secure; SameSite=none",
+            b"foo",
+            b"bar",
+            sameSite="none",
+            secure=True,
+        )
+        self._checkCookie(
+            b"foo=bar; Secure; SameSite=none",
+            b"foo",
+            b"bar",
+            sameSite=b"none",
+            secure=True,
+        )
+
+    def test_addCookieWrongValues(self):
+        """
+        L{http.Request.setCookie} with bad C{sameSite} values.
+        It will set the cookie with not supported samesite value and without a necessary
+        secure attribute for the samesite=none cookie.
```

Comment:
Maybe something like this... as it will **not** accept setting the cookie

```suggestion
        Raises an exception when setting the cookie with not supported samesite value and without a necessary
        secure attribute for the samesite=none cookie.
```